src/flp/types/fixedpoint_l2.rs

```diff
@@ -671,6 +751,16 @@ mod tests {
             vec!(0.25, 0.125, 0.0625)
         );
 
+        // Noise (we only test that the noised vector is not the same as the original)
```

Comment:
To improve test coverage, I think what we would want to do here is be able to fix a seed here form which the gaussian is derived deterministically. That way the test fails if someone fiddles with that code. Consider that if someone changed the implementation of .add_nose() so that some non-random constant was added, we would not detect it.

Reply:
i agree that having an option for seeded noise would be very useful. we are wondering about one thing tho: in production, we have to use proper `getrandom` randomness to generate uniform numbers. this means all our code will be generic over the randomness source somehow, if we add a seeded version. does that make sense, or is there a more reasonable way?

btw, the `opendp` discrete gaussian implementation does not provide seeded sampling, so this would be another reason to adapt it?

Reply:
Yes I think that makes sense.

By the way there is a bigger reason why `opendp` is not desirable for us: It is based on GMP, which is a C library. There is a desire to keep this library pure-Rust, and as far as we can tell so far, there is no other implementation of discrete Gaussian sampler in Rust.